great_expectations/core/factory/checkpoint_factory.py

```diff
@@ -24,6 +24,10 @@
 
 @public_api
 class CheckpointFactory(Factory[Checkpoint]):
+    """
+    Responsible for basic CRUD operations on a context's Checkpoints.
```

Comment:
given that [`ActionContext`](https://docs.greatexpectations.io/docs/reference/api/checkpoint/ActionContext_class) is now a thing, I worry that the mention of "context" here could be ambiguous. It could be more clear to say data context

```suggestion
    Responsible for basic CRUD operations on a Data Context's Checkpoints.
```